packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,502 @@
+    }
+  },
+
+  data () {
+    return {
+      hidePopover: false,
+      tabName: this.getDefaultTabName(),
+      abbreviatedDisplay: this.defaultMessage,
+      fullRangeDisplay: '',
+      selectedCalendarRange: this.defaultCustom,
+      selectedTimeframe: this.timePeriods[0],
+      selectedRange: { start: '', end: '' },
+      modelConfig: {
+        type: 'number'
+      },
+      calendarSelectAttributes: {
+        highlight: {
+          start: { class: 'vcal-day-start' },
+          base: { class: 'vcal-day-base' },
+          end: { class: 'vcal-day-end' },
+          day: { class: 'vcal-day-preselect' }
+        }
+      }
+    }
+  },
+
+  computed: {
+    hasCalendar () {
+      return this.calendarMode !== ''
+    },
+    hasTimePeriods () {
+      return this.timePeriods && this.timePeriods.length
+    },
+    showCalendar () {
+      return this.tabName === 'custom' || !this.hasTimePeriods
+    },
+    hasRelativeTimeframes () {
+      return this.timePeriods.length > 0
+    },
+    submitDisabled () {
+      return this.isRange || this.hasRelativeTimeframes
+        ? !this.selectedRange.start || !this.selectedRange.end
+        : !this.selectedRange.start
+    },
+    hasDefaultCustomValue () {
+      return this.defaultCustom.start !== '' && this.defaultCustom.end !== ''
+    },
+    defaultTimeframe () {
+      return (this.defaultRelative !== undefined)
+        ? this.defaultRelative
+        : null
+    }
+  },
+
+  watch: {
+    // Updates input field's "human" date whenever v-calendar value is touched
+    selectedCalendarRange (newVal) {
+      if (newVal) {
+        let start = ''
+        let end = ''
+
+        // If value is an object, this is a time range. Else, a single date or time value.
+        if (newVal.hasOwnProperty('start')) {
+          start = newVal.start
+          end = newVal.end
+        } else {
+          start = newVal
+        }
+
+        const range = { start, end }
+
+        // Set emitted value when v-calendar selection is made
+        this.selectedRange = {
+          start,
+          end,
+          startISO: new Date(start).toISOString(),
+          endISO: (end !== '') ? new Date(end).toISOString() : end
+        }
+
+        this.abbreviatedDisplay = this.formatDisplayDate(range)
+      }
+    }
+  },
+
+  mounted () {
+    // Select the tab based on incoming defaults
+    if (this.hasDefaultCustomValue) {
+      this.tabName = 'custom'
+    } else if (this.hasRelativeTimeframes) {
+      this.tabName = 'relative'
+    }
+
+    // Set default value to be displayed in the input field
+    if (this.hasDefaultCustomValue) {
+      const range = {
+        start: this.defaultCustom.start,
+        end: this.defaultCustom.end
+      }
+
+      this.abbreviatedDisplay = this.formatDisplayDate(range)
+    } else if (this.defaultTimeframe) {
+      this.changeRelativeTimeframe(this.defaultTimeframe)
+    }
+  },
+
+  methods: {
+    getText (tf) {
+      return tf.timeframeText
+    },
```

Comment:
I don't believe this is being used; can likely be deleted.
```suggestion
```

Reply:
Actually, i needed a simple method to concatenate the prefix (eg: Current, Last, etc) with the timeframeText, so I renamed it `getTimeframeText()` and used it in a couple of places :)